webapp/src/containers/createIncident/index.tsx

```diff
@@ -64,8 +65,21 @@ const UpdateState = () => {
         setChannelOptions([]);
         setApiError(null);
         setValidationError(null);
+        setShowResultPanel(false);
+        setIncidentPayload(null);
+        setSubscriptionPayload(null);
         setShowChannelPanel(false);
+    }, []);
+
+    // Hide the modal and reset the states
+    const hideModal = useCallback(() => {
         dispatch(resetGlobalModalState());
+        resetFieldStates();
+    }, []);
+
+    // Opens incident modal
+    const handleOpenIncidentModal = useCallback(() => {
+        resetFieldStates();
     }, []);
 
     const onShortDescriptionChangeHandle = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
```suggestion
    const handleShortDescriptionChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Rename all the functions like this.